grpc/grpc.go

```diff
@@ -129,6 +198,15 @@ func WithProxyPort(port int) ServerOption {
 	}
 }
 
+// WithRestProxyPort configures the REST Proxy port
+func WithRestProxyPort(port int) ServerOption {
+	return func(o *serverOptions) {
```

Comment:
This is more for knowledge but why return a func that sets it instead of just having the `(o *serverOptions)` in the line above?

Reply:
This is generally a pattern for exposing configurations parameters in go. Admittedly, this particular option doesn't do much, but there are cases where the config does some more complex checks and uses some internal values that is doesn't want to expose to the code using the module.